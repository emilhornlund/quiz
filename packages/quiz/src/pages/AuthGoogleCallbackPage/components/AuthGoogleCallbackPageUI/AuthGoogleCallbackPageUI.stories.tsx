import type { Meta, StoryObj } from '@storybook/react'
import { withRouter } from 'storybook-addon-remix-react-router'

import AuthGoogleCallbackPageUI from './AuthGoogleCallbackPageUI'

const meta = {
  title: 'Pages/AuthGoogleCallbackPage',
  component: AuthGoogleCallbackPageUI,
  decorators: [withRouter],
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof AuthGoogleCallbackPageUI>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {},
} satisfies Story
