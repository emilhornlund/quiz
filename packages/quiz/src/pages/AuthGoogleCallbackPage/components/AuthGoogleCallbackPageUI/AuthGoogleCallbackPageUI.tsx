import React, { FC } from 'react'

import { LoadingSpinner, Page, Typography } from '../../../../components'

const AuthGoogleCallbackPageUI: FC = () => {
  return (
    <Page>
      <Typography variant="subtitle" size="small">
        Hold tight—calling in the Google cavalry!
      </Typography>
      <Typography variant="text" size="small">
        We’re doing the secret handshake with Google. Almost there…
      </Typography>
      <LoadingSpinner />
    </Page>
  )
}

export default AuthGoogleCallbackPageUI
