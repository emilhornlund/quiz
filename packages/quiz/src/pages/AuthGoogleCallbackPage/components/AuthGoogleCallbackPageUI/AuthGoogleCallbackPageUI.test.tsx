import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { describe, expect, it } from 'vitest'

import AuthGoogleCallbackPageUI from './AuthGoogleCallbackPageUI'

describe('AuthGoogleCallbackPageUI', () => {
  it('should render AuthGoogleCallbackPageUI', async () => {
    const { container } = render(
      <MemoryRouter>
        <AuthGoogleCallbackPageUI />
      </MemoryRouter>,
    )

    expect(container).toMatchSnapshot()
  })
})
