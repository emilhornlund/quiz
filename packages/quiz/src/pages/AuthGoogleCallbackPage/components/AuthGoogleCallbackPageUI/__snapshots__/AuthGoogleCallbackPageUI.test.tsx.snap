// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthGoogleCallbackPageUI > should render AuthGoogleCallbackPageUI 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <a
          data-discover="true"
          href="/auth/login"
        >
          Login
        </a>
      </div>
    </div>
    <div
      class="content centerAlign"
    >
      <div
        class="contentWrapper fullWidth"
      >
        <div
          class="contentInner"
        >
          <div
            class="typography subtitle small"
          >
            Hold tight—calling in the Google cavalry!
          </div>
          <div
            class="typography text small"
          >
            We’re doing the secret handshake with Google. Almost there…
          </div>
          <div
            class="loadingSpinnerContainer"
          >
            <div />
            <div />
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
